glom/test/test_match.py

```diff
@@ -0,0 +1,84 @@
+import json
+
+import pytest
+
+from glom import glom, S, Literal, T
+from glom.matching import Match, M, GlomMatchError, And, Or, DEFAULT
+from glom.core import Build, V
+
+
+def _chk(spec, good_target, bad_target):
+    glom(good_target, spec)
+    with pytest.raises(GlomMatchError):
+        glom(bad_target, spec)
+
+def test():
+    _chk(Match(1), 1, 2)
+    _chk(Match(int), 1, 1.0)
+    # test unordered sequence comparisons
+    _chk(Match([int]), [1], ["1"])
+    _chk(Match({int}), {1}, [1])
+    _chk(Match(frozenset({float})), frozenset({}), frozenset({"1"}))
+    with pytest.raises(GlomMatchError):
+        glom([1], Match([]))  # empty shouldn't match
+    glom({"a": 1, "b": 2}, Match({str: int}))
+    glom(2, M == 2)
+    glom(int, M == int)
+    glom(1.0, M > 0)
+    glom(1.0, (M > 0) & float)
+    glom(1.0, (M > 100) | float)
+    glom(1.0, M & (M > 0) & float)
+    glom(1.0, M | (M > 100) | float)
+    # test idiom for enum
+    with pytest.raises(GlomMatchError):
+        glom("c", Match("a"))
+    with pytest.raises(GlomMatchError):
+        glom("c", Match(Or("a", "b")))
+    _chk(Match(M | "a" | "b"), "a", "c")
```

Comment:
Another thing that popped into mind, `M` probably doesn't have any direct comparison usage for callability, so we could make that a shortcut for `Match` construction, maybe?

Reply:
I was thinking maybe the name of `M` should actually change to `C` for "chainable comparator" or something

Basically, `M` and `Match` aren't necessarily connected in usage or implementation even though they arose close to each other

I think keeping `M` and `Match` distinct is really good because `M` is useful inside other schemes/dialects, and also it eliminates a `Literal` vs `Spec` problem

Reply:
but, I definitely had the exact same thoughts around -- should the public constructor for `M` be an alias for `Match()`?

Reply:
(and have not at all rejected that idea)